sympy/sets/ordinals.py

```diff
@@ -0,0 +1,131 @@
+from sympy.core import Basic
+
+def reverse_bisect_right(a, x, lo=0, hi=None):
+    """Return the index where to insert item x in list a, assuming a is sorted in descending order
+
+    Essentially, the function returns number of elements in a which are >= than x.
+    >>> from sympy.sets.ordinals import reverse_bisect_right
+    >>> a = [8, 6, 5, 4, 2]
+    >>> reverse_bisect_right(a, 5)
+    3
+    >>> a[:reverse_bisect_right(a, 5)]
+    [8, 6, 5]
+    """
+    if lo < 0:
+        raise ValueError('lo must be non-negative')
+    if hi is None:
+        hi = len(a)
+    while lo < hi:
+        mid = (lo+hi)//2
+        if x > a[mid]: hi = mid
+        else: lo = mid+1
+    return lo
+
+class Ordinal(Basic):
+    """
+    Represents ordinals in Cantor normal form.
+
+    Internally, this class is just a list of lists (known
+    as "terms").
+    In single terms [a, b], a represents power to the ordinal and b represents multiplying factor
+    Examples
+    ========
+    >>> from sympy.sets import Ordinal, Ordinals
+    >>> Ordinals.w  # doctest: +SKIP
+    w
+    >>> a = Ordinals.w
+    >>> a.is_limit_ordinal
+    True
+    >>> Ordinal([[3,2]])  # doctest: +SKIP
+    {w**3}*2
+    >>> Ordinal([[5,1],[3,2]])  # doctest: +SKIP
+    w**5 + {w**3}*2
+    """
+    def __new__(cls, terms):
+
+        obj = super(Ordinal, cls).__new__(cls, *terms)
+        obj.terms = terms
```

Comment:
It is not necessary to save `terms` as `Basic` will save them and make them available as `args`. In fact, it is not necessary to define `__new__` at all if nothing else need be done.